pronto/static/js/entry/annotations.js

```diff
@@ -415,10 +412,16 @@ const annotationEditor = {
         this.textareaText = null;
         this.textFormatted = null;
     },
-    open: function (annID, text) {
+    open: function (annID, text, references) {
         const element = document.getElementById(annID);
+        const rePub = /\[cite:(PUB\d+)\]/gi;
         let segment;
 
+        text = text.replaceAll(rePub, (match, pubID) => {
+            const pub = references.get(pubID);
+            return `[cite:${pub.pmid}]`
```

Comment:
Return original match if `pubID` not found:

```suggestion
            if (references.has(pubID)) {
                const pub = references.get(pubID);
                return `[cite:${pub.pmid}]`    
            }
            
            return match; 
```